.github/workflows/ci.yaml

```diff
@@ -0,0 +1,30 @@
+name: Cargo Build & Test
+
+on:
+  push: 
+    branches: 
+      - master
+  pull_request:
+    branches:
+      - master
+
+env: 
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_and_test:
+    name: light client project - latest
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toolchain:
+          - stable
+
+    steps:
+      - uses: actions/checkout@v3
+      - run: rustup update ${{ matrix.toolchain }} && rustup default ${{ matrix.toolchain }}
+      - run: cargo clippy
+      - run: cargo fmt -- --check 
+      - run: cargo build --verbose
```

Comment:
if we run test, then I think we can skip build -- we will save some cycles / time.